pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -509,3 +529,595 @@ def reconstructing_fractional_octave_bands(
+                order=self._order)
+
+        # initialize superclass
+        super().__init__(coefficients, self._sampling_rate)
+        self.comment = (
+            f"Second order section 1/{num_fractions} fractional octave band"
+            "filter of order {order}")
+
+    def __repr__(self):
+        """Nice string representation of class instances"""
+        return (f"Energy-preserving fractional "
+                f"octave filter bank with {self.n_bands} "
+                f"bands between {self.frequency_range[0]} and "
+                f"{self.frequency_range[1]} Hz with "
+                f"{self.sampling_rate} Hz sampling rate and "
+                f"{self.num_fractions} per octave.")
+
+    def __eq__(self, other):
+        """Check for equality of two objects."""
+        return not DeepDiff(self.__dict__, other.__dict__)
+
+    @property
+    def num_fractions(self):
+        """Get the number of fractions per octave"""
+        return self._num_fractions
+
+    @property
+    def frequency_range(self):
+        """Get the frequency range of the filter bank in Hz"""
+        return self._frequency_range
+
+    @property
+    def nominal_frequencies(self):
+        """Get the IEC center frequencies of the (fractional)
+         octave bands in Hz"""
+        return self._nominal_frequencies
+
+    @property
+    def center_frequencies(self):
+        """Get the exact center frequencies of the (fractional)
+        octave bands in Hz"""
+        return self._center_frequencies
+
+    @property
+    def cutoff_frequencies(self):
+        """Get the cutoff frequencies of the (fractional)
+        octave bands in Hz"""
+        return self._cutoff_frequencies
+
+    @property
+    def n_bands(self):
+        """Get the number of bands in the filter bank"""
+        return len(self._center_frequencies)
+
+    def _get_coefficients(
+            self,
+            sampling_rate,
+            num_fractions,
+            frequency_range,
+            order):
+        """Calculate the second order section filter
+        coefficients of a fractional octave band filter bank.
+
+        Parameters
+        ----------
+        sampling_rate : int
+            The sampling rate in Hz.
+        num_fractions : int
+            The number of bands an octave is divided into. Eg.,
+            1 refers to octave bands and 3 to third octave bands.
+        frequency_range : array, tuple
+            The lower and upper frequency limits.
+        order : integer, optional
+            Order of the Butterworth filter.
+
+
+        Returns
+        -------
+        sos : array, float
+            Second order section filter coefficients with shape (.., 6)
+
+        Notes
+        -----
+        This function uses second order sections of butterworth filters for
+        increased numeric accuracy and stability.
+        """
+
+        f_crit = fractional_octave_frequencies(
+            num_fractions, frequency_range, return_cutoff=True)[2]
```

Comment:
Examplary for the remainder of this method:
```suggestion
        f_crit = fractional_octave_frequencies(
            self.num_fractions, self.frequency_range, return_cutoff=True)[2]
```